growattServer/__init__.py

```diff
@@ -521,32 +876,45 @@ def plant_info(self, plant_id):
         """
         Get basic plant information with device list.
         """
-        response = self.session.get(self.get_url('newTwoPlantAPI.do'), params={
+        response = self.session.get(self.__get_url('newTwoPlantAPI.do'), params={
             'op': 'getAllDeviceListTwo',
             'plantId': plant_id,
             'pageNum': 1,
             'pageSize': 1
         })
 
-        data = json.loads(response.content.decode('utf-8'))
-        return data
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise Exception('Failed to get plant info')
 
-    def get_plant_settings(self, plant_id):
+    def get_all_devices(self, plant_id):
         """
-        Returns a dictionary containing the settings for the specified plant
+        Get basic plant information with device list.
+        """
+        response = self.session.get(self.__get_url('newTwoPlantAPI.do'), 
+                                     params={'op': 'getAllDeviceList',                                
+                                             'plantId': plant_id,
+                                             'language': 1})
 
-        Keyword arguments:
-        plant_id -- The id of the plant you want the settings of
+        if response.status_code == 200:
+            return response.json()['deviceList']
+        else:
+            raise Exception('Failed to get device data')
 
-        Returns:
-        A python dictionary containing the settings for the specified plant
+    def get_energy_data(self, plant_id):
         """
-        response = self.session.get(self.get_url('newPlantAPI.do'), params={
-            'op': 'getPlant',
-            'plantId': plant_id
-        })
-        data = json.loads(response.content.decode('utf-8'))
-        return data
+        Get the energy data used in the 'Plant' tab in the phone
+        """
+        response = self.session.post(self.__get_url('newTwoPlantAPI.do'), 
+                                     params={'op': 'getUserCenterEnertyDataByPlantid'}, 
```

Comment:
Is this a typo from us or from Growatt? `Enerty`

Reply:
It is a Growatt typo :-).